diff --git a/node_modules/expo-server/build/cjs/vendor/http.js b/node_modules/expo-server/build/cjs/vendor/http.js
index 4ffaac4..c3bf8f9 100644
--- a/node_modules/expo-server/build/cjs/vendor/http.js
+++ b/node_modules/expo-server/build/cjs/vendor/http.js
@@ -98,6 +98,9 @@ function convertHeaders(requestHeaders) {
     return headers;
 }
 async function respond(res, expoRes) {
+    if (res.writableEnded || res.destroyed) {
+        return;
+    }
     res.statusMessage = expoRes.statusText;
     res.statusCode = expoRes.status;
     if (typeof res.setHeaders === 'function') {
diff --git a/node_modules/expo-server/build/mjs/vendor/http.js b/node_modules/expo-server/build/mjs/vendor/http.js
index be9397c..a075bfe 100644
--- a/node_modules/expo-server/build/mjs/vendor/http.js
+++ b/node_modules/expo-server/build/mjs/vendor/http.js
@@ -90,6 +90,9 @@ export function convertHeaders(requestHeaders) {
     return headers;
 }
 export async function respond(res, expoRes) {
+    if (res.writableEnded || res.destroyed) {
+        return;
+    }
     res.statusMessage = expoRes.statusText;
     res.statusCode = expoRes.status;
     if (typeof res.setHeaders === 'function') {
